tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -50,18 +54,70 @@ private SpanMessagePackFormatter()
             _processIdValueBytes = processId > 0 ? MessagePackSerializer.Serialize(processId) : null;
         }
 
+        // IMessagePackFormatter<ArraySegment<Span>> is deprecated, use IMessagePackFormatter<TraceMessageChunk>
+        public int Serialize(ref byte[] bytes, int offset, ArraySegment<Span> value, IFormatterResolver formatterResolver)
+        {
+            var traceContext = value.Array![value.Offset].Context.TraceContext;
+            var traceChunk = new TraceChunkModel(value, traceContext);
+            return Serialize(ref bytes, offset, traceChunk, formatterResolver);
+        }
+
+        public int Serialize(ref byte[] bytes, int offset, TraceChunkModel value, IFormatterResolver formatterResolver)
```

Comment:
Ideally we should use `in` in both the signature and callsites for the `TraceChunkModle` as it's a large `readonly struct`, to avoid defensive copies. Same goes for `SpanModel`

Reply:
Can I do that even though this method implements `IMessagePackFormatter<TraceChunkModel>.Serialize(...)`? The interface declaration does not have the `in` parameter modifier.

Reply:
_Sigh_ you're right, how annoying

Reply:
I added overloads with `in` and made the non-`in` explicit interface implementations.